azurelinuxagent/common/protocol/wire.py

```diff
@@ -72,7 +73,7 @@ def __init__(self, endpoint):
             raise ProtocolError("WireProtocol endpoint is None")
         self.client = WireClient(endpoint)
 
-    def detect(self):
+    def detect(self, goalstate_properties=GoalStateProperties.default_properties()):
```

Comment:
No need to expose the properties in this method

Reply:
This has been removed